test/test_serialization.py

```diff
@@ -318,6 +321,36 @@ def __reduce_ex__(self, proto):
                     "size is inconsistent with indices"):
                 y = torch.load(f)
 
+    def test_serialization_sparse_csr_invalid(self):
+        x = torch.zeros(3, 3)
+        x[1][1] = 1
+        x = x.to_sparse_csr()
+
+        class TensorSerializationSpoofer(object):
+            def __init__(self, tensor):
+                self.tensor = tensor
+
+            def __reduce_ex__(self, proto):
+                invalid_crow_indices = self.tensor.crow_indices().clone()
+                invalid_crow_indices[0] = 3
+                return (
+                    torch._utils._rebuild_sparse_tensor,
+                    (
+                        self.tensor.layout,
+                        (
+                            invalid_crow_indices,
+                            self.tensor.col_indices(),
+                            self.tensor.values(),
+                            self.tensor.size())))
+
+        with tempfile.NamedTemporaryFile() as f:
+            torch.save({"spoofed": TensorSerializationSpoofer(x)}, f)
+            f.seek(0)
+            with self.assertRaisesRegex(
+                    RuntimeError,
+                    "rebuilding sparse tensor for layout torch.sparse_csr"):
```

Comment:
What is the full error message  here?

Reply:
```
NotImplementedError: rebuilding sparse tensor for layout torch.sparse_csr
```

Reply:
Why is this NotImplementedError? Shouldn't this be a runtime error because you can't reconstruct this Tensor because the saved Tensor is invalid?

Reply:
Sure, let's change that in a follow up PR.

Reply:
https://github.com/pytorch/pytorch/pull/89553 fixes this, the error message is now
```python
`crow_indices[..., 0] == 0` is not satisfied.
```
for CSR and BSR tensors.